core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -537,6 +537,11 @@ object SparkSubmit extends CommandLineUtils {
       }
     }
 
+    // SPARK-21568 ConsoleProgressBar should be enabled only in shells
```

Comment:
I'm not a fan of comments like this. It's just repeating what the code is doing.

Instead, can you change the command to mention that it's ok to do this because the user configuration is loaded later in this method and will override this value if set?